### ISSUE TITLE
`imports_granularity = "Module"` breaks code which imports the same identifier twice with two different names

### ISSUE THREAD

**Issue by glittershark** [rust-lang/rustfmt#5224]
`rustfmt.toml`:
```toml
edition = "2018" # not sure if this is necessary for repro
imports_granularity = "Module"
```
`main.rs` (pre formatting):
```rust
use std::collections::HashMap;
use std::collections::HashMap as HM;

fn f(_: &HM<i32, i32>) {}
fn main() {
  let x = HashMap::new();
  f(&x);
}
```

`main.rs` (post formatting):
```rust
use std::collections::HashMap;

fn f(_: &HM<i32, i32>) {}
fn main() {
    let x = HashMap::new();
    f(&x);
}
```

removing the `HashMap as HM` alias causes the code to no longer compile.

**Comment by ytmimi** [comment]
Thanks for reaching out.

By the way, what version of rustfmt are you using? I believe this has been fixed in master by #5209.

Using ``rustfmt 1.4.38-nightly (8b0b213c 2022-01-29)`` I get the following when reformatting your snippet using your specified configuration options:

```rust
use std::collections::{HashMap, HashMap as HM};

fn f(_: &HM<i32, i32>) {}
fn main() {
    let x = HashMap::new();
    f(&x);
}
```

**Comment by calebcartwright** [comment]
Thanks for sharing but closing as a duplicate. It hasn't been released yet but the fix will hit nightly the next time we run a sync
